### PR TITLE
Document substantive changes since Rec as candidate amendments

### PR DESCRIPTION
**Pull request by dontcallmedom**
This pull request is a proposed approach to document the various substantive changes we've brought to the spec since it was published as a Recommendation to make it acceptable as part of the process to amend W3C recommendations.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/pull/2713.html" title="Last updated on Sep 27, 2022, 10:43 AM UTC (cb8fa58)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2713/3d94785...cb8fa58.html" title="Last updated on Sep 27, 2022, 10:43 AM UTC (cb8fa58)">Diff</a>

### CODE REVIEW

**Comment by dontcallmedom** [comment]
The process to add a candidate change with this approach is to add a description of the change based on the following template:
```html
<div class=correction id="change-prXXXX">
<p><span class=title><!-- Description that can be used a summary of the change --></span>
   is a candidate correction - 
  see <a href="https://github.com/w3c/webrtc-pc/pull/XXXX">its associated pull request</a>.</p>
</div>
```
and inserts it in the document close to where the substantive changes were made.

If several distant sections are concerned, further instance of that notice can be inserted by simply adding an empty `<div>` of the form:
```html
<div class=correction id="change-prXXXX-N"></div>
```
where N is the incremented value of the number of these notices.

The ReSpec pre-processing script then takes care of adding the numbered "Candidate Correction" markers, and of listing and linking them in the "Candidate Amendments" appendix.

**Comment by dontcallmedom** [comment]
@alvestrand @caribouW3 @plehegar  I have now taken another stab at this that fulfills the process requirements as I understand them from @plehegar while still reducing the editing burden substantially (although whether it reduces it enough is something I'm looking for @alvestrand feedback).

The result of this work is that:
* the default editors draft view shows the latest content of the spec unchanged https://mir.w3.org/~dom/webrtc-pc/webrtc.html (except for the listing of amendments in appendix)
* when switching to the REC view of the document, it shows in-line candidate corrections, defaulting to showing both current & future, with a toggle to show only the "current" or  "future" view, à la https://mir.w3.org/~dom/webrtc-pc/webrtc.html?specStatus=REC#dictionary-rtcconfiguration-members

I've tried documenting the required steps for integrating an amendment in the draft in `amendments.md` - these steps technically only have to be followed when getting ready to publish a Rec with candidate or proposed amendments (although doing it as we merge substantive pull requests is also possible); my expectation is that this process can be put on the shoulders of the staff contacts rather than editors, since these Rec updates aren't something we expect to do very often in general. From having done it with the current changes, I don't think they're particularly challenging to follow, and certainly less tricky than marking up diffs at the phrase level.

**Comment by dontcallmedom** [comment]
For illustration purposes, I've gone through the exercise of what it would take to import transferable data channels from WebRTC extensions to the main spec as a candidate addition: https://github.com/dontcallmedom/webrtc-pc/pull/2 - https://github.com/dontcallmedom/webrtc-pc/commit/aaed6afab1c7fab1b11d0525e33aff5e9b993e60 is the work needed on top of the spec work to markup the changes as Rec-ready amendments

The same mechanism could be used to bring that section in the editors draft and mark it up as "addition under investigation" there (completely independently of the Rec amendments process), as an alternative to maintaining it in the separate webrtc-extensions repo.

File: amendments.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+* Identify the `id` of the element that wraps the amended content (or alternatively, add a container and/or an `id` to base-rec.html and the relevant block in the editors draft)
+* Add an array under that `id` as top level key in `amendments.json`, or complete the existing array if that `id` is already used
+* Add an an object to that array with keys:
+  * `description`: an HTML description of what the change is,
+  * `dittype` (optional): defaults to "modify" (only the content of the block has changed); can also be "append" (in which case the added blocks in the editors draft should have the class `add-to-`*id*)
```
Comment (alvestrand) [inline amendments.md:5]: dittype->difftype?

File: amendments.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  * `pr`: the number of the pull request where the amendment was merged (or an array thereof),
+  * `type`: ["correction"](https://www.w3.org/2021/Process-20211102/#candidate-correction) or ["addition"](https://www.w3.org/2021/Process-20211102/#candidate-addition),
+  * `status`: ["candidate"](https://www.w3.org/2021/Process-20211102/#candidate-amendment) or ["proposed"](https://www.w3.org/2021/Process-20211102/#last-call-review),
+  * `id` (an incremented number, possibly tying together several changes made across different sections)
+
+Limitations:
+* does not allow to handle different type of changes ("modification" vs "append") in a single section - treating all of them as modification should suffice in such a situation
+* does not allow to handle different type of amendment ("correction" vs "addition") or different status of amendment ("candidate" vs "proposed") in a single section - the only alternative is to find a lower granularity section should that be needed
+
```
[the remark this answers is not in the document]
Comment (dontcallmedom) [reply amendments.md:14]: the changes for section id "dictionary-rtcconfiguration-members" in the pull request illustrates a section with two changes. In general, you don't have to mark them up, you just have to describe the changes in amendments.json, and the script takes care of showing the current rec vs the proposed new text.

**Comment by dontcallmedom** [comment]
@plehegar have you had a chance to check if the results match what you expect would be acceptable for an updated Rec?

**Comment by dontcallmedom** [comment]
@plehegar indicated offline that this approach should pass muster, so I'm now marking it as ready for review & merging, which will help completing upcoming pull requests with additional substantive changes to document them with this mechanism.

**Comment by alvestrand** [comment]
Note: the generated version seemed to have a number of respec errors. Those probably need to be hunted down during the merge process.

**Comment by dontcallmedom** [comment]
I've rebased to main and updated the sample, which removes the said respec errors (the ones that remain are "expected").

**Comment by dontcallmedom** [comment]
I've updated this with all the latest PR that got merged; it would be great to merge this to ensure new substantive changes can be directly annotated using this mechanism, and to experiment with annotating less mature changes as well (as mentioned in #2770

**Comment by dontcallmedom** [comment]
merging as agreed with the chairs

**Pull request merged**